zboxcore/sdk/renameworker.go

```diff
@@ -90,25 +98,31 @@ func (req *RenameRequest) renameBlobberObject(blobber *blockchain.StorageNode, b
 }
 
 func (req *RenameRequest) ProcessRename() error {
-	numList := len(req.blobbers)
-	objectTreeRefs := make([]fileref.RefEntity, numList)
-	req.wg = &sync.WaitGroup{}
-	req.wg.Add(numList)
-	for i := 0; i < numList; i++ {
+	num := len(req.blobbers)
+	objectTreeRefs := make([]fileref.RefEntity, num)
+
+	wait := make(chan RenameResult, num)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(num)
+
+	for i := 0; i < num; i++ {
 		go func(blobberIdx int) {
-			defer req.wg.Done()
-			refEntity, err := req.renameBlobberObject(req.blobbers[blobberIdx], blobberIdx)
-			if err == nil {
-				req.consensus.Done()
-				req.maskMU.Lock()
-				objectTreeRefs[blobberIdx] = refEntity
-				req.maskMU.Unlock()
-				return
+			defer wg.Done()
+			refEntity, err := req.renameBlobberObject(req.blobbers[blobberIdx])
+
+			if err != nil {
+				l.Logger.Error(err.Error())
+			}
+
+			wait <- RenameResult{
+				BlobberIndex: blobberIdx,
+				FileRef:      refEntity,
+				Renamed:      err == nil,
```

Comment:
Seems the RenameResult and CopyResult could be refactored to ProcessResult that with the `Succeed bool` field to indicate whether the err is nil, or just have the `Err error` field that let the caller to check if success. Then we don't need to define result struct respectively for each process type.

Reply:
fixed